addon/components/frost-toggle.js

```diff
@@ -53,45 +58,44 @@ export default Component.extend(SpreadMixin, PropTypeMixin, FrostEventsProxyMixi
   },
 
   getDefaultProps () {
+    const falseLabel = this.get('falseLabel')
+    const trueLabel = this.get('trueLabel')
+    const validTypes = ['string', 'number', 'boolean']
+
     return {
       // options
       disabled: false,
-      _falseLabel: this.get('falseLabel') !== undefined &&
-      (typeOf(this.get('falseLabel') === 'string') || typeOf(this.get('falseLabel') === 'number'))
-                ? this.get('falseLabel') : false,
+      onToggle () {},
       size: 'medium',
-      _trueLabel: this.get('trueLabel') !== undefined &&
-      (typeOf(this.get('trueLabel') === 'string') || typeOf(this.get('trueLabel') === 'number'))
-                ? this.get('trueLabel') : true
+
+      // state
+      _falseLabel: validTypes.includes(typeOf(falseLabel)) ? falseLabel : false,
+      _trueLabel: validTypes.includes(typeOf(trueLabel)) ? trueLabel : true
     }
   },
 
   // == Computed Properties ===================================================
 
+  @readOnly
   @computed('trueValue', '_trueLabel')
-  // FIXME: jsdoc
-  // TODO: make computed property readOnly
-  _trueValue (trueValue, _trueLabel) { // eslint-disable-line
-    return trueValue || _trueLabel
+  _trueValue (trueValue, _trueLabel) {
+    return (trueValue === undefined) ? _trueLabel : trueValue
   },
 
+  @readOnly
   @computed('falseValue', '_falseLabel')
-  // FIXME: jsdoc
-  // TODO: make computed property readOnly
-  _falseValue (falseValue, _falseLabel) { // eslint-disable-line
-    return falseValue || _falseLabel
+  _falseValue (falseValue, _falseLabel) {
+    return (falseValue === undefined) ? _falseLabel : falseValue
```

Comment:
Same obviously - likely we can't do anything about these comments, but maybe we can make an LTS issue pointing to this PR so that we can cleanup in the next release